compiler/circle-execution-plan/src/ExecutionPlanner.cpp

```diff
@@ -74,6 +101,65 @@ void ExecutionPlanner::make_execution_plan()
   settings->set(luci::UserSettings::Key::ExecutionPlanGen, true);
 }
 
+void ExecutionPlanner::create_json_allocation_file(const std::string &json_path)
+{
+  Json::Value main_tree;
+  Json::Value segments_node;
+  Json::Value allocations_node;
+
+  uint32_t alive_till_max = 0;
+
+  // Find max dealloc value to assign to nodes with node_not_assigned value
+  for (const auto elem : _dealloc_node)
+  {
+    if (alive_till_max < elem and elem != node_not_assigned)
+      alive_till_max = elem;
+  }
+
+  for (uint32_t i = 0; i < _ordered_nodes.size(); ++i)
+  {
+    auto current_node_it = std::find_if(
+      _alloc_node_inform_vector.begin(), _alloc_node_inform_vector.end(),
+      [i](const AllocationNodeInformation &x) { return x.node_num == i and (not x.is_temp); });
+
+    auto circle_node = loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]);
+
+    create_allocation_node(allocations_node, *current_node_it, alive_till_max, circle_node);
+
+    // Create and add temporary tensors for current nodes
+    if (_offsets[i].size() > 1)
+    {
+      for (auto &alloc_node_inform : _alloc_node_inform_vector)
+      {
+        if (alloc_node_inform.node_num == i and alloc_node_inform.is_temp)
+        {
+          create_allocation_node(allocations_node, alloc_node_inform, alive_till_max, circle_node);
+        }
+      }
+    }
```

Comment:
Why iterate over nodes, then over `alloc_node_inform_vector`?
Is it possible to iterate only over alloc_node_inform_vector and find nodes by index?

Reply:
@ai-moiseev 
Does order of nodes in file affects visualization?
Current code adds allocations in rpo, what I propose adds nodes in arbitrary order.

Reply:
Rewrite it, thank you